compiler/codegen/CForLoop.cpp

```diff
@@ -154,7 +180,19 @@ GenRet CForLoop::codegen()
                                                FNAME("condition"));
 
     // Create the conditional branch
-    info->builder->CreateCondBr(condValue1, blockStmtBody, blockStmtEnd);
+    llvm::Instruction* endLoopBranch = info->builder->CreateCondBr(condValue1, blockStmtBody, blockStmtEnd);
+
+    if(fNoVectorize == false && isOrderIndependent())
+    {
+        llvm::MDNode* llvmLoopMetadata = addLoopVectorizationHint(endLoopBranch);
+        for(auto it = blockStmtBody->begin(); it != blockStmtBody->end(); it++)
```

Comment:
I think this code might only add parallel loop access metadata to simple loops.
What if we have
``` chapel
var sum = 0;
for i in vectorizeOnly(1..100) with (+ reduce sum) {
  if i > 10 then
     sum += i;
  else
     sum += 1;
}
```

? Won't this loop over blockStmtBody instructions not visit those instructions "inside" the conditional?